### PR TITLE
Make snapshots include actual strings

### LINKED ISSUE #167: Snapshots don't include actual strings
Generated snapshots don't include the actual strings from translation.json; instead they include just the key name (when using i18next-react's t() functions) or just the <Trans> component wrapper (when using i18next-react's Trans component). This means changes in copy/text won't be captured by snapshots, which makes it harder to catch unintentional changes.

### PR DESCRIPTION
**Pull request by kalvinwang**
This PR makes snapshots include actual strings from translation.json, for both t() functions and <Trans> components. Currently, snapshots only include key names or Trans components with no actual strings, so changes in copy/text aren't captured by snapshots.

I updated all the i18n/testing libraries as part of debugging this, so I've run npm update in general, since I'll need to do a bunch of manual testing for this PR anyway.

The code could definitely be streamlined, but given the circumstances I think this is sufficient.

===

Resolves #167

- [x] Snapshots updated, if output HTML/JS has changed (`npm run test:update-snapshots`)
- [ ] WILL DO BEFORE MERGE TO MASTER: changes reviewed on mobile using devtools, if output HTML/CSS has changed
- [ ] Automated tests added, if new functionality added
- [ ] Documentation (e.g. READMEs) updated, if relevant
- [ ] WILL DO BEFORE MERGE TO MASTER: IE11 and Edge compatibility confirmed via manual testing in Browserstack, if new libraries added or newish JS/HTML/CSS features used for the first time in this codebase
- [ ] Accessibility & performance audit run using Google Lighthouse, if major changes made

### CODE REVIEW

File: src/client/components/TabbedContainer/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -39,7 +39,7 @@ exports[`<TabbedContainer /> renders vertical navigation bar 1`] = `
               onKeyDown={[Function]}
               to="/guide/benefits"
             >
-              tabTitles.0
+              Start here
```
Comment (kalvinwang) [inline src/client/components/TabbedContainer/__snapshots__/index.test.js.snap:42]: wooo

File: src/client/components/TabPaneContent1/__snapshots__/index.test.js.snap [hunk 2]
```diff
@@ -1,80 +1,157 @@
...
-      </strong>
+exports[`<TabPaneContent1 /> renders the TabPaneContent component 1`] = `
+<TabPaneContent1>
+  <div>
+    <h3>
+      <Trans
+        i18nKey="tab1-header1"
+        t={[Function]}
+      >
+        Proof of citizenship or authorization to work in the United States
```
Comment (kalvinwang) [inline src/client/components/TabPaneContent1/__snapshots__/index.test.js.snap:11]: wooooo

File: src/client/test-helpers/renderTransContent.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+function renderTransContent(Component, props = {}) {
+  const wrapper = mount(
+    <I18nextProvider i18n={i18n}>
+      <Component {...props} />
+    </I18nextProvider>
+  );
+
+  // Filter out the very verbose I18nextProvider component wrapper
+  return wrapper.find(Component);
```
Comment (sharonwarner) [inline src/client/test-helpers/renderTransContent.js:14]: Interesting!

File: .eslintrc [hunk 4]
```diff
@@ -45,8 +45,6 @@
     "react/jsx-fragments": ["error", "element"],
     "react-hooks/rules-of-hooks": "error",
     "react-hooks/exhaustive-deps": "warn",
-    "sort-imports": "error",
```
Comment (kalvinwang) [inline .eslintrc:48]: this is already in a separate pr for approval

File: public/locales/en/translation.json [hunk 5]
```diff
@@ -111,7 +111,7 @@
   "tab2-p3": "<strong>Restart your old claim</strong>. If your UI application was approved within the past 12 months and you stopped receiving benefits, you don't need to submit a new application. If your claim is more than 12 months old, you will need to file a new claim.",
   "tab3-header1": "1. We'll confirm your application and weekly benefit amount (WBA) via email and postal mail",
   "tab3-list1=item1": "<strong>You should receive the confirmation of your application within two weeks via postal mail.</strong> We'll let you know what your WBA is, based on our wage records. For an initial estimate of your WBA, use our <4>UI benefit calculator</4>.",
-  "tab3-list1-item3": "You may receive an email confirming that you were automatically registered in UI Online. This allows you to fully access information about your claim and to certify for continued benefits online.",
+  "tab3-list1-item2": "You may receive an email confirming that you were automatically registered in UI Online. This allows you to fully access information about your claim and to certify for continued benefits online.",
```
Comment (idealisms) [inline public/locales/en/translation.json:114]: Good catch!

**Pull request merged**